adapter/policy/handler/handler_test.go

```diff
@@ -86,8 +98,75 @@ func OidcClientGenerator(name string, id string, url string) v1.OidcClient {
 	return GetOidcClient(GetOidcClientSpec(name, id, url), GetObjctMeta(), GetTypeMeta())
 }
 
+func OidcClientWithRef(name string, id string, url string, ref v1.ClientSecretRef) v1.OidcClient {
+	return GetOidcClient(
+		v1.OidcClientSpec{
+			ClientName:name,
+			ClientID: id,
+			DiscoveryURL: url,
+			ClientSecret: "",
+			ClientSecretRef: ref,
+		}, GetObjctMeta(), GetTypeMeta())
+}
+
+func OidcClientNoSecret(name string, id string, url string) v1.OidcClient {
+	return GetOidcClient(
+		v1.OidcClientSpec{
+			ClientName:name,
+			ClientID: id,
+			DiscoveryURL: url,
+		}, GetObjctMeta(), GetTypeMeta())
+}
+
 func TestNew(t *testing.T) {
-	assert.NotNil(t, New(store.New()))
+	assert.NotNil(t, New(store.New(), fake.NewSimpleClientset()))
+}
+
+func TestGetClientSecret(t *testing.T) {
+	testHandler := &CrdHandler{
+		store: store.New(),
+		kubeClient: fake.NewSimpleClientset(),
+	}
+	tests := []struct {
+		obj v1.OidcClient
+		secret string
+	}{
+		{
+			obj: GetOidcClient(GetOidcClientSpec("name", "id", "url"), GetObjctMeta(), GetTypeMeta()),
+			secret: secretFromPlainText,
+		},
+		{
+			obj: OidcClientNoSecret("name", "id", "url"),
+			secret: "",
+		},
+		{
+			obj: OidcClientWithRef("name", "id", "url", v1.ClientSecretRef{Name:"mysecret", Key:"secretKey"}),
+			secret: secretFromRef,
+		},
+		{
+			obj: OidcClientWithRef("name", "id", "url", v1.ClientSecretRef{Name:"mysecret", Key:"invalidKey"}),
+			secret: "",
+		},
+		{
+			obj: OidcClientWithRef("name", "id", "url", v1.ClientSecretRef{Name:"invalidName", Key:"secretKey"}),
+			secret: "",
+		},
+		{
+			obj: GetOidcClient(
+				v1.OidcClientSpec{
+					ClientName:"name",
+					ClientID: "id",
+					DiscoveryURL: "url",
+					ClientSecret:secretFromPlainText,
+					ClientSecretRef: v1.ClientSecretRef{Name:"mysecret", Key: ""}}, GetObjctMeta(), GetTypeMeta()),
+			secret: secretFromPlainText,
+		},
+	}
+	_, _ = testHandler.kubeClient.CoreV1().Secrets("sample").Create(MockKubeSecret())
+	for _, test := range tests {
+		res := testHandler.getClientSecret(&test.obj)
```

Comment:
Lets parallelize these tests like [this](https://github.com/ibm-cloud-security/policy-enforcer-mixer-adapter/blob/development/adapter/strategy/api/api_test.go#L73)